hudi-common/src/main/java/org/apache/hudi/metadata/FileSystemBackedTableMetadata.java

```diff
@@ -84,6 +96,19 @@ public List<String> getAllPartitionPaths() throws IOException {
     return getPartitionPathWithPathPrefixes(Collections.singletonList(""));
   }
 
+  @Override
+  public List<String> getPartitionPathByExpression(List<String> relativePathPrefixes,
```

Comment:
Lets have a standalone test for this method in this class and in HoodieBackedTableMetadata as this is the crux of this PR.

Reply:
Sure, will add tests soon